x-pack/test/functional_with_es_ssl/apps/cases/group2/attachment_framework.ts

```diff
@@ -325,6 +331,88 @@ export default ({ getPageObject, getService }: FtrProviderContext) => {
         });
       });
     });
+
+    describe('Lens visualization as persistable attachment', () => {
+      const myDashboardName = `My-dashboard-${uuidv4()}`;
+      before(async () => {
+        await esArchiver.loadIfNeeded('x-pack/test/functional/es_archives/logstash_functional');
+        await kibanaServer.importExport.load(
+          'x-pack/test/functional/fixtures/kbn_archiver/lens/lens_basic.json'
+        );
+        await common.navigateToApp('dashboard');
+        await dashboard.preserveCrossAppState();
+        await dashboard.clickNewDashboard();
+
+        // adds lens visualization to dashboard, save and return
+        await lens.createAndAddLensFromDashboard({ title: `My lens visualization-${uuidv4()}` });
+
+        await dashboard.waitForRenderComplete();
+        await dashboard.saveDashboard(myDashboardName);
+      });
+
+      after(async () => {
+        await esArchiver.unload('x-pack/test/functional/es_archives/logstash_functional');
+        await kibanaServer.importExport.unload(
+          'x-pack/test/functional/fixtures/kbn_archiver/lens/lens_basic.json'
+        );
+
+        await common.navigateToApp('dashboard');
+        await dashboard.preserveCrossAppState();
+
+        await listingTable.searchForItemWithName(myDashboardName);
+        await listingTable.checkListingSelectAllCheckbox();
+        await listingTable.clickDeleteSelected();
+      });
+
+      it('adds lens visualization to a new case from dashboard', async () => {
```

Comment:
How about we add the navigation logic to this test too? That way if we add a test ahead of this one that leaves us at a different location, this test won't start failing.

I think these are the lines
```
        await common.navigateToApp('dashboard');
        await dashboard.preserveCrossAppState();
        await dashboard.loadSavedDashboard(myDashboardName);
```